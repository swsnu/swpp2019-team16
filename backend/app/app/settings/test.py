from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
