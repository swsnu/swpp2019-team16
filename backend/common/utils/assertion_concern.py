class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
