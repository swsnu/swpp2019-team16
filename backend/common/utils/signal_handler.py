import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
