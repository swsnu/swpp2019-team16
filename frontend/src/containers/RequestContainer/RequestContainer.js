import React, { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Request from '../../components/Request';
import { withRouter } from 'react-router-dom';
import { requestCarpool } from '../../modules/carpoolRequest';

RequestContainer.propTypes = {};

function RequestContainer({ history }) {
  const dispatch = useDispatch();
  const user = useSelector(user => ({
    user: user.user,
  }));

  //TODO: it should be loaded from DB
  const fromList = ['SNU station', 'Nakseongdae station', 'Nokdu Street'];
  const toList = ['301 building', 'Student Center', 'Dormitory three-way'];
  const minimumPassenger = ['2', '3', '4'];

  const onClickRequest = useCallback(
    ({ userId, from, toString, minimumPassenger }) => {
      dispatch(requestCarpool({ userId, from, toString, minimumPassenger }));
      history.push('/waiting');
    },
    [dispatch, history],
  );

  if (!user) {
    return <div>we are loading user...</div>;
  }

  return (
    <Request
      user={user}
      fromList={fromList}
      toList={toList}
      minimumPassenger={minimumPassenger}
      onClickRequest={onClickRequest}
    />
  );
}

export default withRouter(RequestContainer);
