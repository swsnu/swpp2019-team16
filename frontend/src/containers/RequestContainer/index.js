import RequestContainer from './RequestContainer';
export default RequestContainer;
