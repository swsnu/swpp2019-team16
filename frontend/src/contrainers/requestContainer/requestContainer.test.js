import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import RequestContainer from './requestContainer';
import { renderWithRedux } from '../../test/utils';
import { MemoryRouter, Route } from 'react-router-dom';

jest.mock('../../components/request/request', () =>
  jest.fn(props => <div></div>),
);

describe('<RequestContainer />', () => {
  const state = {
    user: {
      user: {
        id: 1,
      },
    },
  };

  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/request']}>
        <Route component={RequestContainer} path="/request" />
      </MemoryRouter>,
      state,
    );

    expect(container).toMatchSnapshot();
  });
});
