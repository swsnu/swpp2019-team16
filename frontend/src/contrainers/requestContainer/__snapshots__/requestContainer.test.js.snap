// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestContainer /> SHOULD match with snapshot 1`] = `
<div>
  <div />
</div>
`;
